src/sqlfluff/core/templaters/slicers/tracer.py

```diff
@@ -222,13 +232,16 @@ def slice_info_for_literal(self, length, prefix="") -> RawSliceInfo:
         """
         unique_alternate_id = self.next_slice_id()
         alternate_code = f"\0{prefix}{unique_alternate_id}_{length}"
-        return self.make_raw_slice_info(unique_alternate_id, alternate_code)
+        return self.make_raw_slice_info(
+            unique_alternate_id, alternate_code, inside_block=self.inside_block
+        )
 
     def update_inside_set_or_macro(
```

Comment:
Is this function name still accurate? Or should it now be `update_inside_set_or_macro_or_block`?